### PR TITLE
Feature/imgui polish

### PR DESCRIPTION
**Pull request by Aiden-Lonquist**
- Cleaned up the appearance of the GUI
- Fixed positioning of all windows
- Changed drop down lists to radio buttons
- Added toggle to only show the UI in debug mode
- Added feature to add a script to an entity through the GUI

### CODE REVIEW

**Review by Hevano — DISMISSED** [review]
Looks good to me

File: Engine/src/Engine/EngineUI/GamePanel.h [hunk 1]
```diff
@@ -7,9 +7,15 @@ namespace Engine
...
 	public:
-		void setInitialPosition();
+
+		virtual ~GamePanel() {}
+
 		void show(Window& window);
+
 		glm::vec2 getPosition() { return m_position; };
+
+		const int titleBarHeight = 36;
```
Comment (mikesinclair66) [inline Engine/src/Engine/EngineUI/GamePanel.h]: const should be following the naming convention of VARIABLE_NAME

File: Engine/src/Engine/EngineUI/UserInterface.cpp [hunk 2]
```diff
@@ -10,14 +10,14 @@ namespace Engine
 	std::unordered_map<std::string, ImFont*> UserInterface::s_fonts;
 
 	//define the ImGUI vec4s for each colour
-	ImVec4 UserInterface::red = ImVec4(1.0f, 0.0f, 0.0f, 1.0f);
-	ImVec4 UserInterface::green = ImVec4(0.0f, 1.0f, 0.0f, 1.0f);
-	ImVec4 UserInterface::darkGreen = ImVec4(0.0f, 0.75f, 0.0f, 1.0f);
-	ImVec4 UserInterface::blue = ImVec4(0.0f, 0.0f, 1.0f, 1.0f);
-	ImVec4 UserInterface::grey = ImVec4(0.75f, 0.75f, 0.75f, 1.0f);
+	ImVec4 UserInterface::lightGrey = ImVec4(0.75f, 0.75f, 0.75f, 1.0f);
```
Comment (mikesinclair66) [inline Engine/src/Engine/EngineUI/UserInterface.cpp]: If plausible, these colour variables should be consts and following the corresponding naming conventions.

Comment (mikesinclair66) [reply Engine/src/Engine/EngineUI/UserInterface.cpp]: Can be const ImVec4 btw. Can be complicated implementing cpp variables as const when they're also stored as const in the header.

**Review by mikesinclair66 — CHANGES_REQUESTED** [review]
Looks great, just some things about the const vars.

File: Engine/src/Engine/EngineUI/GamePanel.h (hunk 1 above)
Comment (Aiden-Lonquist) [reply Engine/src/Engine/EngineUI/GamePanel.h]: fixed name

File: Engine/src/Engine/EngineUI/UserInterface.cpp (hunk 2 above)
Comment (Aiden-Lonquist) [reply Engine/src/Engine/EngineUI/UserInterface.cpp]: changed colours to const and updated names

**Review by mikesinclair66 — DISMISSED** [review]

File: Engine/src/Engine/EngineUI/InspectorPanel.cpp [hunk 3]
```diff
@@ -117,12 +135,33 @@ namespace Engine
...
+
 				auto* serializable = &(m_registry->get<SerializableComponent>(m_entityHandle));
 				ImGui::Checkbox("Is serializable", &serializable->serializable);
+
+				setSpacing(spacing);
+
+				ImGui::PopFont();
+			}
+
+			/*if (m_registry->any_of<ScriptComponent>(m_entityHandle))
```
Comment (roughleythomas) [inline Engine/src/Engine/EngineUI/InspectorPanel.cpp]: Please remove if not needed

**Review by roughleythomas — DISMISSED** [review]

**Review by roughleythomas — APPROVED** [review]

**Review by Hevano — APPROVED** [review]
Revisions look great

**Pull request merged**